src/api/Flooq.Api/Controllers/DataFlowController.cs

```diff
@@ -126,30 +127,37 @@ public async Task<ActionResult<DataFlow>> PutDataFlow(Guid? id, DataFlow dataFlo
     }
 
     // POST: api/DataFlow
-    // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
     /// <summary>
     /// Adds a <see cref="DataFlow"/>.
     /// If null, the uuid is automatically created and set.
     /// If null, the status is set to "Disabled".
     /// Even if not null, the field lastEdited will be ignored. Instead, it's automatically created.
     /// </summary>
     /// <param name="dataFlow">The new <see cref="DataFlow"/>.</param>
-    /// <returns>A <see cref="CreatedAtActionResult"/> object that produces a <see cref="StatusCodes.Status201Created"/> response.</returns>
+    /// <returns>
+    /// A <see cref="CreatedAtActionResult"/> object that produces a <see cref="StatusCodes.Status201Created"/> response
+    /// or <see cref="ConflictResult"/> if the <see cref="DataFlow"/> already exists.
+    /// </returns>
     [HttpPost]
     [Authorize("write")]
     public async Task<ActionResult<DataFlow>> PostDataFlow(DataFlow dataFlow)
     {
-      if (DataFlowExists(dataFlow.Id))
-      {
-        _dataFlowMetricsService.IncrementBadRequestCount();
-        return BadRequest();
-      }
-
       dataFlow.UserId = GetCurrentUserId();
       dataFlow.LastEdited = DateTime.UtcNow;
-
+      
       _dataFlowService.AddDataFlow(dataFlow);
-      await _dataFlowService.SaveChangesAsync();
+      try
+      {
+        await _dataFlowService.SaveChangesAsync();
+      }
+      catch (Exception)
```

Comment:
The problem is that we get a `System.ArgumentException` if the dataflow already exists. I therefore made it general, but I know this might not be the proper way. @MrF3lix what do you think of that?

Reply:
For me it's ok to use `Exception` here. Important for me is that the status code returned is correct and by adding the additional check if it exists this is alright for me.